src/server/config.js

```diff
@@ -1,48 +1,47 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
-var config = {};
+/**
+ * The Configuration model manages the configuration properties for the simulation.
+ * These properties are mapped with the options provided for the simulation, and
+ * are cross to any other instance involved in the simulation.
+ * @constructor
+ */
+function Configuration() {
+    this._config = {};
+}
 
-// module properties
 [
     { name: 'debugHostHandlers', optional: true },
     { name: 'forcePrepare' },
     { name: 'liveReload' },
-    { name: 'platform' },
-    { name: 'platformRoot' },
-    { name: 'projectRoot', single: true },
-    { name: 'server', optional: true },
     { name: 'simHostOptions' },
     { name: 'simulationFilePath' },
     { name: 'telemetry', optional: true },
     { name: 'touchEvents', optional: true },
     { name: 'xhrProxy', optional: true }
 ].forEach(function (prop) {
-    Object.defineProperty(module.exports, prop.name, {
+    Object.defineProperty(Configuration.prototype, prop.name, {
         get: function () {
-            return getValue(prop.name, prop.optional);
+            return getValue(this, prop.name, prop.optional);
         },
         set: function (value) {
-            setValue(prop.name, value, prop.single);
+            setValue(this, prop.name, value, prop.single);
         }
     });
 });
 
-function setValue(prop, value, single) {
-    if (single && config.hasOwnProperty(prop)) {
+function setValue(instance, prop, value, single) {
+    if (single && instance._config.hasOwnProperty(prop)) {
         throw new Error('Can\'t reinitialize ' + prop);
     }
-    config[prop] = value;
+    instance._config[prop] = value;
 }
 
-function getValue(prop, optional) {
-    if (!config.hasOwnProperty(prop) && !optional) {
+function getValue(instance, prop, optional) {
+    if (!instance._config.hasOwnProperty(prop) && !optional) {
```

Comment:
Just a word of caution: if you remove `_config` and inherit from `Object.create(null)`, then `hasOwnProperty` will not work. You can, though, instead rely on the fact that the properties you define on `Configuration.prototype` itself are non-configurable (and specify it explicitly).